plugins/components/acceleration/Acceleration.hpp

```diff
@@ -171,8 +166,15 @@ namespace Components {
 
             std::cout << "forward: " << forwardVal << " backward: " << backwardVal << " right: " << rightVal << " left: " << leftVal << std::endl;
 
-            std::unique_ptr<Components::Acceleration> pos = engine.newComponent<Components::Acceleration>(static_cast<int32_t>(forwardVal), static_cast<int32_t>(backwardVal), static_cast<int32_t>(rightVal), static_cast<int32_t>(leftVal));
-            engine.getRegistry().componentManager().addComponent<Components::Acceleration>(to, std::move(pos));
+            std::unique_ptr<Components::Acceleration> acceleration = engine.newComponent<Components::Acceleration>(static_cast<int32_t>(forwardVal), static_cast<int32_t>(backwardVal), static_cast<int32_t>(rightVal), static_cast<int32_t>(leftVal));
+            Components::IComponent *rawComponent = acceleration.get();
+            engine
+                .getRegistry()
+                .componentManager()
+                .addComponent<Components::Acceleration>(to, std::move(acceleration));
+            std::cerr << "Acceleration component added to entity: " << to << std::endl;
+            engine.updateComponent(to, rawComponent->getId(), rawComponent->serialize());
+            std::cerr << "Acceleration component updated" << std::endl;
             std::cout << std::endl;
```

Comment:
Is it necessary to get the raw pointer ? Couldn't you just do `engine.updateComponent(to, acceleration->getId(), acceleration->serialize());`

Also is the debug left on purpose ?

Reply:
This comment applies to all the components modified : Acceleration, Collider, Controllable, Damage, DeathRange, Gun, Health, Position, Scale, SpriteID, Type, Velocity and Visible

Reply:
Since we are moving newly created components instead the componentManager, we loose the unique ptr. So we get the raw ptr before hand in order to still be able to call the getId() and serialize() functions.

Reply:
Oh okay I see, very well.

Reply:
A good fix would be to create a temporary helper function to handle this temporary problem

Reply:
It has been done !